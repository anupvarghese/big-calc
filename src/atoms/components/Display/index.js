// @flow

import React from 'react';
import { css } from 'glamor';
import Input from '../../input';
import C from '../../../constants';

type displayT = {
  value: number,
};

const Display = ({ value }: displayT) => {
  return (
    <div {...css({ display: 'flex', flexDirection: 'row' })}>
      <Input
        baseStyle={{
          minHeight: '5rem',
          flexBasis: '100%',
          background: C.BLACK,
          borderColor: `${C.BLACK}`,
          fontSize: '1.8rem',
          paddingLeft: '1rem',
          paddingRight: '1rem',
          textAlign: 'right',
          fontFamily: '-apple-system, BlinkMacSystemFont, sans-serif',
          color: 'white',
        }}
        type="text"
        disabled
        value={value}
      />
    </div>
  );
};

export default Display;
