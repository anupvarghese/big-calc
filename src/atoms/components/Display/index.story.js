import React from 'react';

import { storiesOf } from '@storybook/react';

import Display from './';

storiesOf('Display').add('show display', () => <Display value={1234} />);
