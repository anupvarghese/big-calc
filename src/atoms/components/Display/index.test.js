import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-glamor-react';

import Display from './';

expect.addSnapshotSerializer(serializer);

describe('Display', () => {
  it('renders Display', () => {
    const display = renderer.create(<Display />).toJSON();
    expect(display).toMatchSnapshot();
  });
});
