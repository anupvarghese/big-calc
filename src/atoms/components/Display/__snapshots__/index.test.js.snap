// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders Display 1`] = `
.css-9v06sd,
[data-css-9v06sd] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
}

.css-1to9pp2,
[data-css-1to9pp2] {
  min-height: 5rem;
  flex-basis: 100%;
  background: #292929;
  border-color: #292929;
  font-size: 1.8rem;
  padding-left: 1rem;
  padding-right: 1rem;
  text-align: right;
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  color: white;
  -webkit-flex-basis: 100%;
}

<div
  data-css-9v06sd=""
>
  <input
    data-css-1to9pp2=""
    disabled={true}
    type="text"
    value={undefined}
  />
</div>
`;
